lib/src/main/java/vmath/Mat4.java

```diff
@@ -12,6 +12,13 @@ public Mat4() {
         };
     }
 
+    public Mat4(float[] m) {
+        if (m.length != 16) {
+            throw new IllegalArgumentException("Array length must be 16 for a 4x4 matrix.");
+        }
+        this.m = m.clone();
```

Comment:
I think we've missed the NaN checks in the constructor and the appropriate test method for that like in [here](https://github.com/toksaitov/vmath/pull/53/files).